app/src/main/java/be/scri/fragments/LanguageSettingsFragment.kt

```diff
@@ -161,6 +165,32 @@ class LanguageSettingsFragment : Fragment() {
         return list
     }
 
+    private fun getLayoutRecyclerViewData(language: String): List<SwitchItem> {
+        val sharedPref = requireActivity().getSharedPreferences("app_preferences", Context.MODE_PRIVATE)
+        val list = mutableListOf<SwitchItem>()
+
+        list.add(
+            SwitchItem(
+                isChecked = sharedPref.getBoolean("disable_accent_character_$language", false),
+                title = getString(R.string.app_settings_keyboard_layout_disable_accent_characters),
+                description = getString(R.string.app_settings_keyboard_layout_disable_accent_characters_description),
+                action = { disableAccentCharacter(language) },
+                action2 = { enableAccentCharacters(language) },
+            ),
+
```

Comment:
Instead of this could u replace it with 

```
        when (language) {
            "German" -> {
                list.add(
                    SwitchItem(
                        isChecked = sharedPref.getBoolean("disable_accent_character_$language", false),
                        title = getString(R.string.app_settings_keyboard_layout_disable_accent_characters),
                        description = getString(R.string.app_settings_keyboard_layout_disable_accent_characters_description),
                        action = { disableAccentCharacter(language) },
                        action2 = { enableAccentCharacters(language) },
                    ),
                )
            }
            "Swedish" -> {
                list.add(
                    SwitchItem(
                        isChecked = sharedPref.getBoolean("disable_accent_character_$language", false),
                        title = getString(R.string.app_settings_keyboard_layout_disable_accent_characters),
                        description = getString(R.string.app_settings_keyboard_layout_disable_accent_characters_description),
                        action = { disableAccentCharacter(language) },
                        action2 = { enableAccentCharacters(language) },
                    ),
                )
            }
            "Spanish" -> {
                list.add(
                    SwitchItem(
                        isChecked = sharedPref.getBoolean("disable_accent_character_$language", false),
                        title = getString(R.string.app_settings_keyboard_layout_disable_accent_characters),
                        description = getString(R.string.app_settings_keyboard_layout_disable_accent_characters_description),
                        action = { disableAccentCharacter(language) },
                        action2 = { enableAccentCharacters(language) },
                    ),
                )
            }
        }

```

I had added the accent character option to Functionality earlier Apologies. Since it is not required for all languages to disable accent characters You could replace the menu option to be generated based on languages as in the code above.